File: freecodecamp/daily-code-challenge/2026-04-07.py
def palindrome_locator(s):
    # Check palindrome
    if s != s[::-1]:
        return "none"

    length = len(s)
    mid = length // 2

    # Odd length
    if length % 2 != 0:
        return s[mid]

    # Even length
    return s[mid - 1] + s[mid]

File: freecodecamp/daily-code-challenge/2026-04-07.js
function palindromeLocator(str) {
    // Check palindrome
    if (str !== str.split('').reverse().join('')) {
        return "none";
    }

    let len = str.length;
    let mid = Math.floor(len / 2);

    // Odd length
    if (len % 2 !== 0) {
        return str[mid];
    }

    // Even length
    return str[mid - 1] + str[mid];
}